src/Symfony/Component/Config/Definition/BaseNode.php

```diff
@@ -50,6 +56,39 @@ public function __construct(?string $name, NodeInterface $parent = null, string
         $this->pathSeparator = $pathSeparator;
     }
 
+    /**
+     * Register possible (dummy) values for a dynamic placeholder value. Matching configuration values will be processed
+     * with a provided value, one by one. After a provided value is successfully processed the configuration value is
+     * returned as is, thus preserving the placeholder.
+     */
+    public static function setPlaceholder(string $placeholder, array $values): void
+    {
+        if (!$values) {
+            throw new \InvalidArgumentException('At least one value must be provided.');
+        }
+
+        self::$placeholders[$placeholder] = $values;
+    }
+
+    /**
+     * Set a common prefix for dynamic placeholder values. Matching configuration values will be skipped from being
+     * processed and are returned as is, thus preserving the placeholder. An exact match provided by {@see setPlaceholder()}
+     * might take precedence.
+     */
+    public static function setPlaceholderUniquePrefix(string $prefix): void
+    {
+        self::$placeholderUniquePrefix = $prefix;
+    }
+
+    /**
+     * Reset all current placeholders available.
+     */
+    public static function resetPlaceholders(): void
+    {
+        self::$placeholderUniquePrefix = null;
+        self::$placeholders = array();
```

Comment:
Having static properties like this is very dangerous as we need to make sure that they are reset properly. I can see the `finally` calls here and there to reset this, but is there is any other way? Avoiding static properties would be great.

Reply:
It's not that dangerous: placeholders are unique strings so there is not really any way to mess up with them, even if someone forgets to reset them.
About removing the static state, that's really hard...
BUT, @ro0NL we should make these methods `@internal` for sure.

Reply:
I think we can settle indeed, i dont see a elegant workaround to avoid static currently, that is without overhauling the config component itself :) 

Now marked internal.